### ISSUE TITLE
Missing cmake while compiling xmr-stak.

### ISSUE THREAD

**Issue by papampi** [papampi/nvOC_miners#23]
> X
> Compiling  xmr-stak
>  This could take a while ...
> Submodule 'XMR_Stak/src' (https://github.com/fireice-uk/xmr-stak) registered for path 'src'
> Cloning into '/home/m1/NVOC/3_0/miners/XMR_Stak/src'...
> remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
> remote: Enumerating objects: 156, done.
> remote: Counting objects: 100% (156/156), done.
> remote: Compressing objects: 100% (79/79), done.
> remote: Total 80 (delta 64), reused 11 (delta 0), pack-reused 0
> Unpacking objects: 100% (80/80), done.
> From https://github.com/fireice-uk/xmr-stak
>  * branch            c0ab1734332d6472225d8ac7394f6fcba71aabc9 -> FETCH_HEAD
> Submodule path 'src': checked out 'c0ab1734332d6472225d8ac7394f6fcba71aabc9'
> Warning: you are leaving 1 commit behind, not connected to
> any of your branches:
> 
>   c0ab173 Merge pull request #1610 from fireice-uk/dev
> 
> If you want to keep it by creating a new branch, this may be a good time
> to do so with:
> 
>  git branch <new-branch-name> c0ab173
> 
> HEAD is now at 4e72408 Merge pull request #1958 from fireice-uk/dev
> ~/NVOC/3_0/miners/XMR_Stak/src ~/NVOC/3_0/miners
> ~/NVOC/3_0/miners/XMR_Stak/src/build ~/NVOC/3_0/miners/XMR_Stak/src ~/NVOC/3_0/miners
> ./nvOC_miner_update.sh: line 607: cmake: command not found
> make: *** No rule to make target 'install'.  Stop.
> ~/NVOC/3_0/miners/XMR_Stak/src ~/NVOC/3_0/miners
> cp: cannot stat '/home/m1/NVOC/3_0/miners/XMR_Stak/src/build/bin/xmr-stak': No such file or directory
> cp: cannot stat '/home/m1/NVOC/3_0/miners/XMR_Stak/src/build/bin/*.so': No such file or directory
> ~/NVOC/3_0/miners
> 
> Finished compiling xmr-stak
> 

I tried others choice in the miner update script and all the script untar correctly but all have problemzs to compile. All indicates same errors messages : 

> HEAD is now at 4e72408 Merge pull request #1958 from fireice-uk/dev
> ~/NVOC/3_0/miners/XMR_Stak/src ~/NVOC/3_0/miners
> ~/NVOC/3_0/miners/XMR_Stak/src/build ~/NVOC/3_0/miners/XMR_Stak/src ~/NVOC/3_0/miners
> ./nvOC_miner_update.sh: line 607: cmake: command not found
> make: *** No rule to make target 'install'.  Stop.
> ~/NVOC/3_0/miners/XMR_Stak/src ~/NVOC/3_0/miners
> cp: cannot stat '/home/m1/NVOC/3_0/miners/XMR_Stak/src/build/bin/xmr-stak': No such file or directory
> cp: cannot stat '/home/m1/NVOC/3_0/miners/XMR_Stak/src/build/bin/*.so': No such file or directory
> ~/NVOC/3_0/miners

all the miners i choose returned the same erros.

For xmr-stak, i had to install dependencies: 
 sudo apt install libmicrohttpd-dev libssl-dev cmake build-essential libhwloc-dev

_Originally posted by @cryptatometrader44 in https://github.com/papampi/nvOC_miners/pull/22#issuecomment-431630859_

**Comment by papampi** [comment]
@LuKePicci 
Please have a look see if you can find the problem.

**Comment by LuKePicci** [comment]
For sure we can disable the http api in compilation command line such that http-dependencies are not needed, in future we could handle compilation dependencies in pm manifest.
Here the problem is just the absence of cmake, I see no other issues in that output. The warning about some commits left behind depend on multiple things, the xmr-repo graph and the fact the submodule reference is still pointing to 2.4.7 in our 3.0-miners branch. None of them should cause problems although, since we made the update script capable of checking out what we specified inside itself..
